### DISCUSSION TITLE
diagram.net extension

### DISCUSSION THREAD

**Discussion by Adams-Scott** [standardnotes/forum#2634] [Suggestions]
**Have you read the [Longevity Statement](https://standardnotes.org/longevity)?**
Yes - this may be in conflict with #1, feel free to say #2.

**Is your feature request related to a problem? Please describe.**
There isn't a way to create diagrams and edit it in Standard Notes. I could use an editor to embed a picture, but it wouldn't be editable in the future.

**Describe the solution you'd like**
I would like an extension for diagrams.net or baked in support with one of the existing editors.
https://www.diagrams.net/integrations.html

**Describe alternatives you've considered**
I've seen another request for Mermaid. I gave that a thumbs up.

**Additional context**
A lot of self hosted Wiki's have support for diagrams.net. In wiki.js, you have a button to to create a diagram that brings you to a nice visual editor much like Visio. When you save it, it saves on your wiki page as a code block. When you render your page, you get a nice looking diagram!

I'm not a programmer and I really have no idea how feasible this is. It looks like there are options to host the data and web tool yourself, or just host data and use their instance in an iframe.

**Comment by JaspalSuri** [comment]
Thanks for the feedback! It looks like it uses Java so it might not work with our apps, but I'll pass it along to the dev team for further consideration. 🙂

**Comment by Adams-Scott** [comment]
Great, thanks!

**Comment by shagr4th** [comment]
You can already have an editable diagram, with just 2 extra steps. Create a new one on app.diagrams.net, click on File->Embed->SVG, and copy/paste the generated SVG content in your note. You can alter your diagram by clicking on the image, modify it (bottom right icon), and repeat the file->embed, copy/paste steps
